apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[resourceType]/index.vue

```diff
@@ -0,0 +1,185 @@
+<script setup lang="ts">
+import { IFilter, ISchemaMetaData, ITableMetaData } from "~/interfaces/types";
+const config = useRuntimeConfig();
+const route = useRoute();
+const router = useRouter();
+
+const pageSize = 10;
+const resourceName: string = route.params.resourceType as string;
+const schemaName = route.params.schema.toString();
+const metadata = await fetchMetadata(schemaName);
+
+const tableMetaDataFinderResult = metadata.tables.find(
+  (t: ITableMetaData) =>
+    t.id.toLocaleLowerCase() === resourceName.toLocaleLowerCase()
+);
+
+const tableMetaData = computed(() => {
+  if (tableMetaDataFinderResult) {
+    return tableMetaDataFinderResult;
+  } else {
+    throw new Error(`Table metadata not found for ${resourceName}`);
+  }
+});
+
+const resourceType = tableMetaData.value.id;
+
+const resourceAgg: string = resourceType + "_agg";
+
+const externalSchemaNames: string[] = extractExternalSchemas(metadata);
+const externalSchemas = await Promise.all(
+  externalSchemaNames.map(fetchMetadata)
+);
+const schemas = externalSchemas.reduce(
+  (acc: Record<string, ISchemaMetaData>, schema) => {
+    acc[schema.name] = schema;
+    return acc;
+  },
+  { [schemaName]: metadata }
+);
+
+const description = tableMetaData.value?.descriptions?.[0]?.value;
+
+let activeName = ref("detailed");
+let filters: IFilter[] = reactive([
+  {
+    title: `Search in ${resourceName}`,
+    columnType: "_SEARCH",
+    search: "",
+    searchTables: [],
+    initialCollapsed: false,
+  },
+]);
+
+const currentPage = ref(1);
+
+function setCurrentPage(pageNumber: number) {
+  router.push({ path: route.path, query: { page: pageNumber } });
+  currentPage.value = pageNumber;
+}
+
+if (route.query?.page) {
+  const queryPageNumber = Number(route.query?.page);
+  currentPage.value =
+    typeof queryPageNumber === "number" ? Math.round(queryPageNumber) : 1;
+}
+let offset = computed(() => (currentPage.value - 1) * pageSize);
+
+watch(filters, () => {
+  setCurrentPage(1);
+});
+
+// build resource query for cards
+
+const fields = buildRecordListQueryFields(resourceType, schemaName, schemas);
+
+const query = computed(() => {
+  return `
+  query ${resourceType}($filter:${resourceType}Filter, $orderby:${resourceType}orderby){
+    ${resourceType}(limit: ${pageSize} offset: ${offset.value} filter:$filter  orderby:$orderby) {
+      ${fields}
+    }
+    ${resourceType}_agg (filter:$filter){
+        count
+    }
+  }
+  `;
+});
+const orderby = {};
+let search = computed(() => {
+  // @ts-ignore
+  return filters.find((f) => f.columnType === "_SEARCH").search;
+});
+const filter = computed(() => buildQueryFilter(filters, search.value));
+
+console.log("query: ", query.value);
```

Comment:
log

Reply:
suggest to leave ( for now) , could look into logging framework